### PR TITLE
Cap setuptools even if installed packages are ignored

### PR DESCRIPTION
**Pull request by astrojuanlu**
Continuation of #8711, see discussion at #8775.

### CODE REVIEW

**Comment by astrojuanlu** [comment]
The checks failure is fixed by #8774.

The fact that no tests failed makes me think that this code path is not being tested at all? I will try to add one.

**Comment by astrojuanlu** [comment]
Tests passing :+1:

File: readthedocs/rtd_tests/tests/test_doc_building.py [hunk 1]
```diff
@@ -1210,8 +1211,9 @@ def assertArgsStartsWith(self, args, call):
         with each element of args.
         """
         args_mock, _ = call
-        for arg, arg_mock in zip(args, args_mock):
+        for arg, arg_mock in zip_longest(args, args_mock):
```
Comment (astrojuanlu) [inline readthedocs/rtd_tests/tests/test_doc_building.py:1214]: Otherwise, `zip` was discarding the elements of the longer list

**Review by humitos — APPROVED** [review]
Awesome! Thanks Juanlu for taking care of this!

I'd like to deprecate "use system package" eventually, tho. 😄

**Comment by astrojuanlu** [comment]
> I'd like to deprecate "use system package" eventually, tho. 😄

Me too :)

**Pull request merged**